code/drasil-gool/lib/GOOL/Drasil/State.hs

```diff
@@ -468,7 +469,7 @@ callMapTransClosure = over callMap tClosure
         traceCalls :: Map QualifiedName [QualifiedName] -> [QualifiedName] -> 
           [QualifiedName]
         traceCalls _ [] = []
-        traceCalls cm (c:cs) = nub $ c : traceCalls cm (nub $ cs ++ 
+        traceCalls cm (c:cs) = toList . fromList $ c : traceCalls cm ( cs ++ 
```

Comment:
I think that if you remove the existing `toList . fromList`s, this PR will be in a mergeable state, no? I think we should do that, since you're now going through the `nub`d lists noted by `stan` and figuring out which ones really need a `nub` and which ones really want to be a different type.

Reply:
For this example, you can but for nubSort `toList . fromList` is used to remove all duplicates from the list
nubSort :: Ord a => [a] -> [a]
nubSort = toList . fromList . sort

Reply:
Yep, we can just revert that one to `nub` as it was before this PR. This way, we minimize the changes in the PR and put it in an immediately merge-able state with just the few changes from lists to `Set`s.

After this PR, we can figure out which `nub`s involve a list that should be a `Set` or not, and go from there. However, since `toList . fromList` sorts lists, I think that is also a clue that we can also change this area to using a `Set` (since a `Set` in Haskell is ordered already).